src/map/lua/lua_baseentity.cpp

```diff
@@ -14422,6 +14422,80 @@ bool CLuaBaseEntity::deleteRaisedChocobo()
     return true;
 }
 
+void CLuaBaseEntity::setMannequinPose(uint16 itemID, uint8 race, uint8 pose)
+{
+    TracyZoneScoped;
+    // Find the item in the player inventory and update the extra values
+    auto*  PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity);
+    CItem* PItem = nullptr;
+
+    if (PChar)
+    {
+        for (uint8 i = 0; i < CONTAINER_ID::MAX_CONTAINER_ID; ++i)
+        {
+            if (auto slot = PChar->getStorage(i)->SearchItem(itemID); slot != ERROR_SLOTID)
+            {
+                auto* item = PChar->getStorage(i)->GetItem(slot);
+                if (item && item->getID() == itemID && item->isMannequin())
+                {
+                    PItem = item;
+                    break;
+                }
+            }
+        }
+    }
+
+    auto* PMannequin = dynamic_cast<CItemFurnishing*>(PItem);
+
+    if (PMannequin)
+    {
+        PMannequin->setMannequinRace(race);
+        PMannequin->setMannequinPose(pose);
+
+        // Include the update right into the database
+        char        extra[sizeof(PMannequin->m_extra) * 2 + 1];
+        const char* fmtQuery = "UPDATE char_inventory  \
+                               SET extra = '%s' \
+                               WHERE charid = %u AND itemId = %u;";
+
+        sql->EscapeStringLen(extra, (const char*)PMannequin->m_extra, sizeof(PMannequin->m_extra));
+        if (sql->Query(fmtQuery, extra, PChar->id, PMannequin->getID()) == SQL_ERROR)
```

Comment:
If the check for PChar above fails, we'll dereference a null ptr here

Reply:
Would it get there?  PMannequin only gets cast if there's a PItem, and PItem only gets assigned if PChar succeeds.  
I can make a change, just not sure what the best way would be.

Reply:
CLuaBaseEntity has more than just CCharEntity supported.  By using dynamic_cast, if it is called on other subclasses, it will be null by result, and therefore should have a null check.  Instead of the assignment, I would suggest a conditional block containing the code:

`if (auto PChar = dynamic_cast<CCharEntity*>(m_pBaseEntity))`